src/Network/Network/VirtualNetworkGateway/StopAzVirtualNetworkGatewayPacketCaptureCommand.cs

```diff
@@ -0,0 +1,122 @@
+        [ValidateNotNullOrEmpty]
+        public PSVirtualNetworkGateway InputObject { get; set; }
+
+        [Parameter(
+            ParameterSetName = "ByResourceId",
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Azure resource ID of the VirtualNetworkGateway where packet capture to be started.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "SAS URL packet capture on virtual network gateway.")]
+        public string SasUrl { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+            PSVirtualNetworkGateway existingVirtualNetworkGateway = null;
+            if (ParameterSetName.Equals("ByInputObject"))
+            {
+                existingVirtualNetworkGateway = this.InputObject;
+                this.ResourceGroupName = this.InputObject.ResourceGroupName;
+                this.Name = this.InputObject.Name;
+            }
+            else
+            {
+                if (ParameterSetName.Equals("ByResourceId"))
+                {
+                    var parsedResourceId = new ResourceIdentifier(ResourceId);
+                    Name = parsedResourceId.ResourceName;
+                    ResourceGroupName = parsedResourceId.ResourceGroupName;
+                }
+
+                existingVirtualNetworkGateway = this.GetVirtualNetworkGateway(this.ResourceGroupName, this.Name);
+            }
+
+            if (existingVirtualNetworkGateway == null)
+            {
+                throw new PSArgumentException(Properties.Resources.ResourceNotFound, "Virtual Network Gateway");
+            }
+
+            VpnPacketCaptureStopParameters parameters = new VpnPacketCaptureStopParameters();
+            if (this.SasUrl != null)
+            {
+                parameters.SasUrl = SasUrl;
+            }
+            else
+            {
+
+            }
+            WriteVerbose(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
+            PSVirtualNetworkGatewayPacketCaptureResult output = new PSVirtualNetworkGatewayPacketCaptureResult()
+            {
+                Name = existingVirtualNetworkGateway.Name,
+                ResourceGroupName = existingVirtualNetworkGateway.ResourceGroupName,
+                Tag = existingVirtualNetworkGateway.Tag,
+                ResourceGuid = existingVirtualNetworkGateway.ResourceGuid,
+                Location = existingVirtualNetworkGateway.Location,
+
+            };
+            output.StartTime = DateTime.UtcNow;
+            var result = this.VirtualNetworkGatewayClient.StopPacketCapture(this.ResourceGroupName, this.Name, parameters);
```

Comment:
Need to implement ShouldProcess

Reply:
implemented ShouldProcess for all the commands. Thanks